services/QuillLMS/app/controllers/auth/classlink_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'uri'
+
+module Auth
+  class ClasslinkController < ApplicationController
+    around_action :force_writer_db_role, only: [:classlink]
+    CLIENT_ID = 'c17370377633256aee86a52043b96d2913e27b36df24'
```

Comment:
Let's extract `CLIENT_ID`, `OAUTH_TOKEN_ENDPOINT` and `client_secret` into a new file `config/initializers/auth/classlink` per the pattern with existing providers `auth/{canvas, clever, google}`

That way these are loaded once.